scripts/extensions/auto-tagging-widget/src/ImageTaggingComponent/ImageTaggingComponent.tsx

```diff
@@ -0,0 +1,355 @@
+    }
+
+    componentWillUnmount() {
+      this._mounted = false;
+      this.abortController.abort();
+      clearTimeout(this.fetchTimeout);
+    }
+
+    componentDidUpdate() {
+      clearTimeout(this.fetchTimeout);
+      if (
+        this._mounted &&
+        !this.state.isLoading &&
+        !this.isEqualTags(
+          this.formatTags(this.props.data),
+          this.state.prevInput,
+        )
+      ) {
+        this.fetchTimeout = setTimeout(() => {
+          this.runFetchImages();
+        }, 1000);
+      }
+    }
+
+    runFetchImages() {
+      const formattedTags: Array<ITagInput> = this.formatTags(this.props.data);
+      if (!formattedTags || formattedTags.length < 1) {
+        this.setState({
+          selectedImage: null,
+          images: [],
+        });
+        return;
+      }
+      this.setState({ isLoading: true, prevInput: formattedTags }, () => {
+        httpRequestJsonLocal<IImageServerResponse>({
+          abortSignal: this.abortController.signal,
+          method: "POST",
+          path: "/ai_image_suggestions/",
+          payload: {
+            service: "imatrics",
+            item: formattedTags,
+          },
+        })
+          .then((res) => {
+            if (this._mounted) {
+              this.setState({ fetchError: false });
+              try {
+                this.setState({
+                  selectedImage: res.result[0],
+                  images: res.result ?? [],
+                });
+              } catch {
+                this.setState({
+                  selectedImage: null,
+                  images: [],
+                });
+              }
+            }
+          })
+          .catch(() => {
+            this.setState({ fetchError: true });
+          })
+          .finally(() => this._mounted && this.setState({ isLoading: false }));
+      });
+    }
+
+    isEqualTags = (current: Array<ITagInput>, prev: Array<ITagInput>) =>
+      JSON.stringify(current) === JSON.stringify(prev)
+
+    formatTags = (concepts: IServerResponse) => {
+      let res: Array<ITagInput> = [];
+      for (const key in concepts) {
+        if (key === "subject") {
+          concepts[key]?.forEach((concept: ITagBase) => {
+            res.push({
+              title: concept.name,
+              type: "category",
+              pubStatus: true,
+              weight: 1,
+            });
+          });
+        } else if (
+          key === "organisation" ||
+          key === "person" ||
+          key === "event" ||
+          key === "place" ||
+          key === "object"
+        ) {
+          concepts[key]?.forEach((concept: ITagBase) => {
+            res.push({
+              title: concept.name,
+              type: key,
+              pubStatus: true,
+              weight: 1,
+            });
+          });
+        }
+      }
+      return res;
+    }
+
+    handleClickImage = (image: IImage) => {
+      this.setState({ selectedImage: image });
+    }
+
+    handleDragStart = (
```

Comment:
It only accesses `this.props.article` from the class. Move it out of a class and define as a standalone function. Pass `this.props.article` as an argument.

Reply:
Thank you! handleDragStart has now been moved out of the class.